### PR TITLE
feat: seo canonical link support

### PR DESCRIPTION
**Pull request by Sebastian-Haehnlein**
## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] CI-related changes
- [ ] Documentation content changes
- [ ] application / infrastructure changes
- [ ] Other... Please describe:

## What Is the Current Behavior?

Issue Number: ISREST-932

## What Is the New Behavior?

canonical link support

## Does this PR Introduce a Breaking Change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by dhhyi** [comment]
@shauke merge when you are ready :wink:

**Comment by shauke** [comment]
Empty canonical links should probably not be created, e.g. for the `/home` or `/register` or `/contact` pages. Example:
```
<link rel="canonical">
```

For content pages like About Us it seems to work well without specific handling.

**Comment by fmalcher** [comment]
Possibly related: https://github.com/angular/angular/issues/34605

File: src/app/extensions/seo/store/seo/seo.effects.ts [hunk 1]
```diff
@@ -17,12 +20,20 @@ import { SeoActionTypes, SetSeoAttributes } from './seo.actions';
...
   constructor(
     private actions$: Actions,
     private store: Store<{}>,
     private meta: MetaService,
-    private translate: TranslateService
-  ) {}
+    private translate: TranslateService,
+    @Inject(DOCUMENT) private doc: Document
+  ) {
+    this.cannonicalLink = this.doc.createElement('link');
```
Comment (fmalcher) [inline src/app/extensions/seo/store/seo/seo.effects.ts]: should fix the typo "cannonical" => "canonical".
No cannons 💣 😎

Comment (d-koppenhagen) [reply src/app/extensions/seo/store/seo/seo.effects.ts]: You should consider also checking if there is an existing canonical link set (to his may appear when changing a route). In that case you shouldn't just add the link but update the existing one.
I've implemented in a similar way here: https://github.com/d-koppenhagen/d-koppenhagen.de/commit/1fd231a061803dffdb7bae700090e0f18da14cb7

Comment (dhhyi) [reply src/app/extensions/seo/store/seo/seo.effects.ts]: Implementing it here would be even better: https://github.com/fulls1z3/ngx-meta/issues/44

Comment (Sebastian-Haehnlein) [reply src/app/extensions/seo/store/seo/seo.effects.ts]: ngx-meta dont support canonical-links so i will do in our seo.effects

Comment (dhhyi) [reply src/app/extensions/seo/store/seo/seo.effects.ts]: That was my suggestion, to implement it there... 😅

**Pull request merged**